test/ETHStaticAToken.test.ts

```diff
@@ -1,126 +1,175 @@
-  it('allows owner to update accRewards', async () => {
-    await owner.invoke(tokenContract, 'push_acc_rewards_per_token', {
+  it("allows owner to update accRewards", async () => {
+    await owner.invoke(tokenContract, "push_acc_rewards_per_token", {
       block: 1,
       acc_rewards_per_token: {
         high: 0,
         low: 1,
       },
     });
 
-    const {acc_rewards_per_token} = await tokenContract.call('get_acc_rewards_per_token')
+    const { acc_rewards_per_token } = await tokenContract.call(
+      "get_acc_rewards_per_token"
+    );
     expect(acc_rewards_per_token).to.deep.equal({
       high: 0n,
       low: 1n,
-    })
+    });
   });
 
-  it('dissalows non-owner to update accRewards', async () => {
+  it("dissalows non-owner to update accRewards", async () => {
     try {
-      await user1.call(tokenContract, 'push_acc_rewards_per_token', {
+      await user1.call(tokenContract, "push_acc_rewards_per_token", {
         block: 2,
         acc_rewards_per_token: {
           high: 0n,
           low: 100n,
         },
       });
       expect.fail("non-owner was able to update rewards");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('only allows increases in accRewards', async () => {
+  it("only allows increases in accRewards", async () => {
     try {
-      await owner.call(tokenContract, 'push_acc_rewards_per_token', {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
         block: 3,
         acc_rewards_per_token: {
           high: 0,
           low: 0,
-        }
+        },
       });
       expect.fail("accRewards was decreased");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('rejects old block numbers', async () => {
+  it("rejects old block numbers", async () => {
     try {
-      await owner.call(tokenContract, 'push_acc_rewards_per_token', {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
         block: 0,
         acc_rewards_per_token: {
           high: 0,
           low: 2,
-        }
+        },
       });
       expect.fail("accRewards accepted for old block number");
-    } catch (e) {
-    }
+    } catch (e) {}
   });
 
-  it('updatesUserAccRewardsPerToken', async () => {
-    expect.fail("not implemented yet")
+  it("updatesUserAccRewardsPerToken", async () => {
+    try {
+      await owner.call(tokenContract, "push_acc_rewards_per_token", {
+        block: 0,
+        acc_rewards_per_token: {
+          high: 0,
+          low: 2,
+        },
+      });
+      expect.fail("accRewards accepted for old block number");
+    } catch (e) {}
   });
 
-  it('distributes rewards correctly', async () => {
-    expect.fail("not implemented yet")
+  it("claims rewards to user", async () => {
+    try {
+      await user1.call(tokenContract, "claim_rewards", {
+        user: user1.starknetContract.address,
+        recipient: user1.starknetContract.address,
+      });
+
+      const { userPendingRewards } = await tokenContract.call(
```

Comment:
Can we also add tests that explicitly update the accrued rewards and checks for the exact value on the user. And another with two users to make sure the rewards are distributed correctly?

Reply:
done